src/engine/Server.cpp

```diff
@@ -657,9 +657,10 @@ boost::asio::awaitable<void> Server::processQuery(
     // (tsv, csv, octet-stream, turtle).
     auto sendStreamableResponse = [&](MediaType mediaType) -> Awaitable<void> {
       auto responseGenerator = co_await computeInNewThread(
-          [&plannedQuery, &qet, mediaType] {
+          [&plannedQuery, &qet, mediaType, &cancellationHandle] {
```

Comment:
Not really sure why this is wrapped in computeInNewThread. Doesn't really make sense, because this is only for the creation of the generator object

Reply:
Thank you, that is very thoughtful.
Do you mean that rather the rest of the function that does the actually expensive part (computing the result and exporting it) should be on one of the other threads? Maybe you can fix this as part of this PR.)

Reply:
I think the best thing to do here is to just drop the worker threads completely here. The io is supposed to be done on the io_context. When using worker tasks this will just introduce a lot of overhead when constantly passing data around between threads.
I don't remember exactly how I implemented the streaming functionality a couple years back, but depending on the exact implementation it either already is an issue (if the generator is consumed synchronously and blocks), or the control completely lies within boost beast which would mean that the generator has to wait for io in-between each step anyways.

Reply:
I checked and `createOkResponse` ends up spawning a dedicated thread, so it becomes a non-issue when the `computeInNewThread` wrapper is removed.